### PR TITLE
Added extra checks to docs

### PR DESCRIPTION
**Pull request by ciaranRoche**
## Motivation

The following checks where added after the docs where merged:
- ALLOW_BACKUP_ENABLED
- HAS_ENCRYPTION_ENABLED

## Description

The above checks where added to the docs, complete with links to relative external resources.

### CODE REVIEW

File: docs/getting-started/auth-self-defence-checks.adoc [hunk 1]
```diff
@@ -48,6 +48,14 @@ To detect if the app is being run on an emulator the `SecurityCheckType#IS_EMULA
 
 To detect if a device has a lock screen set (with pin, fingerprint, pattern) the `SecurityCheckType#HAS_SCREENLOCK` function can be used.  This function uses Android's link:https://developer.android.com/reference/android/app/KeyguardManager.html[KeyguardManager] class.
 
+==== link:javadoc[ALLOW_BACKUP_ENABLED]
```
Comment (aidenkeating) [inline docs/getting-started/auth-self-defence-checks.adoc]: Are all these links meant to be broken?

Comment (ciaranRoche) [reply docs/getting-started/auth-self-defence-checks.adoc]: Good idea, I  will remove the javadoc links now :+1:

File: docs/getting-started/auth-self-defence-checks.adoc [hunk 2]
```diff
@@ -28,26 +28,36 @@ There are two ways to implement self defence checks: <<Provided Self Defence Che
 
 These are self defence checks provided by the SDK in link:javadoc[`SecurityCheckType`].
 
-==== link:javadoc[IS_ROOTED]
+// TODO: make check titles links to javadoc when javadoc location is available
```
Comment (secondsun) [inline docs/getting-started/auth-self-defence-checks.adoc]: Please don't have todo's in the docs.

Comment (ciaranRoche) [reply docs/getting-started/auth-self-defence-checks.adoc]: Cool, I will remove that now, I also created an issue for it here [#132](https://github.com/aerogear/aerogear-android-sdk/issues/132)

**Comment by TommyJ1994** [comment]
Might just be me, but would it be better to have more of an english description in the header for each check along with the actual check constant? eg. 

Device Lock Detection (SCREEN_LOCK_ENABLED)
Root Access Detection (IS_ROOTED)
Device Encryption Detection (HAS_ENCRYPTION_ENABLED)

LGTM otherwise. :+1:

**Comment by ciaranRoche** [comment]
@TommyJ1994 Mind having a look at the name changes when you get a chance, cheers.

File: docs/getting-started/auth-self-defence-checks.adoc [hunk 3]
```diff
@@ -28,26 +28,34 @@ There are two ways to implement self defence checks: <<Provided Self Defence Che
...
+==== Emulator Detection (IS_EMULATOR)
 
 To detect if the app is being run on an emulator the `SecurityCheckType#IS_EMULATOR` function can be used.  This function uses Android's link:https://developer.android.com/reference/android/os/Build.html[Build] class.
 
-==== link:javadoc[SCREEN_LOCK_ENABLED]
+==== Device Lock Detection (SCREEN_LOCK_ENABLED)
 
 To detect if a device has a lock screen set (with pin, fingerprint, pattern) the `SecurityCheckType#HAS_SCREENLOCK` function can be used.  This function uses Android's link:https://developer.android.com/reference/android/app/KeyguardManager.html[KeyguardManager] class.
 
+==== Device Backup Detection (ALLOW_BACKUP_ENABLED)
```
Comment (TommyJ1994) [inline docs/getting-started/auth-self-defence-checks.adoc]: `App Data Backup Detection` might be a better description.

**Pull request merged**